clientd3d/ownerdrw.c

```diff
@@ -68,23 +68,39 @@ void WindowEndUpdate(HWND hwnd)
  */
 int OwnerListAddItem(HWND hwnd, object_node *obj, int index, Bool combo, Bool quan)
 {
-   char *name, desc[MAXNAME + 15];
-   int pos;
+   std::string raritySuffix;
+   char nameWithSuffix[MAXNAME + 15];
+   char nameWithQuantity[MAXNAME + 15];
+   char nameWithItemInUse[MAXNAME + 15];
```

Comment:
Weird mix of std::string and char[] here.  Can you make them all strings?  Then you don't need to worry about the size of the buffer, etc.

Reply:
Changed all `char` to `(std::string)` as recommended and it seems to work well now.  

However I had to convert the final name string to a `LPCTSTR` `char*` since the ListBox API requires that type of parameter.